synapse/lib/drive.py

```diff
@@ -60,24 +61,32 @@ def getPathNorm(self, path):
 
         return [reqValidName(p.strip().lower()) for p in path]
 
-    def getItemInfo(self, iden):
-        return self._getItemInfo(s_common.uhex(iden))
+    def getItemInfo(self, iden, typename=None):
+        info = self._getItemInfo(s_common.uhex(iden))
+        if typename is not None and info.get('type') != typename:
+            infotype = info.get('type')
+            mesg = f'Drive item has the wrong type. Expected: {typename} got {infotype}.'
+            raise s_exc.TypeMismatch(mesg=mesg)
```

Comment:
Do we want to include `typename` and `infotype` keys here as well?

Reply:
We never do anything with them, so *shrug*.